lib/web_console/middleware.rb

```diff
@@ -79,26 +78,42 @@ def create_regular_or_whiny_request(env)
         whiny_requests ? WhinyRequest.new(request) : request
       end
 
-      def update_re
-        @options[:update_re]
+      def mount_point
+        @options[:mount]
       end
 
-      def binding_change_re
-        @options[:binding_change_re]
+      def repl_sessions_re
+        %r{#{mount_point}/repl_sessions/(?<id>[^/]+)}
       end
 
-      def id_for_repl_session_update(request)
-        if request.xhr? && request.put?
-          update_re.match(request.path_info) { |m| m[:id] }
-        end
+      def id_for_repl_session(request)
+        repl_sessions_re.match(request.path_info) { |m| m[:id] }
+      end
+
+      def define_route(request_method, regex)
+        { method: request_method, regex: regex }
       end
 
-      def id_for_repl_session_stack_frame_change(request)
-        if request.xhr? && request.post?
-          binding_change_re.match(request.path_info) { |m| m[:id] }
+      def session_api
+        @_session_api ||= {
+          update_repl_session: define_route(:put?, %r{#{repl_sessions_re}\z}),
+          change_stack_trace: define_route(:post?, %r{#{repl_sessions_re}/trace\z}),
+        }
+      end
+
+      # Find and call a suitable method for the request with session.
+      def resolve_session_request(request, id)
+        if action = find_action_for_request(session_api, request)
```

Comment:
Can we keep those as before? I don't like the magical send we do here, while we can explicitly call those methods. Sometimes, IMO, explicitness is better than DRY.